src/main/codegen.ml

```diff
@@ -451,12 +506,25 @@ let translate (globals, functions, externs) =
         Llvm.add_case switch_inst (Llvm.const_int base_types.char_t 0) truthy_bb; (* empty << 1 + is_zero == 0 ===> truthy *)
         Llvm.add_case switch_inst (Llvm.const_int base_types.char_t 1) falsey_bb; (* empty << 1 + is_zero == 1 ===> falsey *)
         (ret_val, merge_builder)
+      | UnOp(LogNot, expr) ->
+        let (truth_val, truth_builder) = build_expr old_builder (UnOp(Truthy, expr)) in
+        let the_number = (truth_val => (value_field_index Number)) "the_number" truth_builder in
+        let not_the_number = Llvm.build_fsub (Llvm.const_float base_types.float_t 1.0) the_number "not_the_number" truth_builder in
+        let sp = Llvm.build_struct_gep truth_val (value_field_index Number) "num_pointer" truth_builder in
+        let _ = Llvm.build_store not_the_number sp truth_builder in
+        (truth_val, truth_builder)
+      | UnOp(Neg, expr) ->
+        let vvv = Llvm.const_float base_types.float_t 42.0 in
```

Comment:
I wanted to demonstrate the appropriate usage of debug_print. Was not sure how else to show it in action, but I think it's probably clear now! I will remove - please use this function instead of one-off printf's. The "flags" thing was making us pass certain test cases we shouldn't have been passed so I kept having to move things back out of inputs_regression that should not have ended up there in the first place.